src/hotspot/share/cds/cppVtables.cpp

```diff
@@ -66,19 +66,20 @@
 
 class CppVtableInfo {
   intptr_t _vtable_size;
-  intptr_t _cloned_vtable[1];
+  intptr_t _cloned_vtable[1]; // Pseudo flexible array member.
+  static size_t cloned_vtable_offs() { return offset_of(CppVtableInfo, _cloned_vtable); }
 public:
   static int num_slots(int vtable_size) {
     return 1 + vtable_size; // Need to add the space occupied by _vtable_size;
   }
   int vtable_size()           { return int(uintx(_vtable_size)); }
```

Comment:
There's a bunch of pre-existing weirdness around the type of _vtable_size.  (I think _every_ use involves a
conversion.)  Doing anything about that doesn't really belong in this change, but consider a followup cleanup.

Reply:
Right. Please note that I usually don't touch code in this area. If you would like it to get improved, I suggest filing an RFE and discussing with the CDS folks. My intention is to get rid of UB which is terrible.